src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -419,5 +539,22 @@ private AsnWriter WriteSubjectPublicKeyInfo()
             ImportParameters(key);
             bytesRead = localRead;
         }
+
+        public int GetMaxSignatureSize(DSASignatureFormat signatureFormat)
+        {
+            DSAParameters dsaParameters = ExportParameters(false);
+            int qLength = dsaParameters.Q.Length;
+
+            switch (signatureFormat)
+            {
+                case DSASignatureFormat.IeeeP1363FixedFieldConcatenation:
+                    return qLength * 2;
+                case DSASignatureFormat.Rfc3279DerSequence:
+                    // Add 15 for extra ASN.1 headers
+                    return qLength * 2 + 15;
```

Comment:
If we're using this as the lower bound of what someone has to pass in (at least for the OpenSsl implementation) then maybe it needs to be a more accurate max.

// 30 7E 02 3E 00 [qLength] 02 3E 00 [qlength]
* qLength <= 0x3D (62) ? qLength * 2 + 8;
// 30 81 FE 02 7C 00 [qLength] 02 7C 00 [qLength]
* qLength <= 0x7B (124) ? qLength * 2 + 9;
// 30 82 01 00 02 7F 00 [qLength] 02 7F 00 [qLength]
* qLength <= 0x7E (126) ? qLength * 2 + 10
// 30 82 02 02 02 81 FF 00 [qLength] 02 81 FF 00 [qLength]
* qLength <= 0xFE (254) ? qLength * 2 + 12
// 30 82 FF FE 02 82 7F FD 00 [qLength] 02 82 7F FD 00 [qLength]
* qLength <= 0x7FFC (32764) ? qLength * 2 + 14
// 30 83 02 00 02 02 82 FF FF 00 [qLength] 02 82 FF FF 00 [qLength]
* qLength <= 0xFFFE (65534) ? qLength * 2 + 15
// 30 83 FF FF FE 02 83 7F FF FD 00 [qLength] 02 83 7F FF FD 00 [qLength]
* qLength <= 0x7F_FFFC (8_388_604) ? qLength * 2 + 17
// 30 84 02 00 00 02 02 83 FF FF FF 00 [qLength] 02 83 FF FF FF 00 [qLength]
* qLength <= 0xFF_FFFE (16_777_214) ? qLength * 2 + 18
// 30 84 FF FF FF FE 02 84 7F FF FF FD 00 [qLength] 02 84 7F FF FF FD 00 [qLength]
* qLength <= 0x7FFF_FFFC (2_147_483_644) ? qLength + 20

Beyond here we need the CLR to support bigger arrays.  Probably could have stopped a bit ago :)

Reply:
Also, FWIW, the biggest current legal size of qLength is 32.  (FIPS 186-4 says the biggest legal DSA key size has N=256, and N is the length of q, in bits).